common/type_system/defenum.cpp

```diff
@@ -0,0 +1,154 @@
+/*!
+ * @file defenum.cpp
+ * Parser for the GOAL "defenum" form.
+ * This is used both in the compiler and in the decompiler for the type definition file.
+ */
+
+#include "Enum.h"
+#include "defenum.h"
+#include "deftype.h"
+#include "third-party/fmt/core.h"
+
+namespace {
+const goos::Object& car(const goos::Object* x) {
+  if (!x->is_pair()) {
+    throw std::runtime_error("invalid defenum form");
+  }
+
+  return x->as_pair()->car;
+}
+
+const goos::Object* cdr(const goos::Object* x) {
+  if (!x->is_pair()) {
+    throw std::runtime_error("invalid defenum form");
+  }
+
+  return &x->as_pair()->cdr;
+}
+
+bool is_type(const std::string& expected, const TypeSpec& actual, const TypeSystem* ts) {
+  return ts->tc(ts->make_typespec(expected), actual);
+}
+
+bool integer_fits(s64 in, int size, bool is_signed) {
+  switch (size) {
+    case 1:
+      if (is_signed) {
+        return in >= INT8_MIN && in <= INT8_MAX;
+      } else {
+        return in >= 0 && in <= UINT8_MAX;
+      }
+    case 2:
+      if (is_signed) {
+        return in >= INT16_MIN && in <= INT16_MAX;
+      } else {
+        return in >= 0 && in <= UINT16_MAX;
+      }
+    case 4:
+      if (is_signed) {
+        return in >= INT32_MIN && in <= INT32_MAX;
+      } else {
+        return in >= 0 && in <= UINT32_MAX;
+      }
+    case 8:
+      return true;
+    default:
+      assert(false);
+      return false;
+  }
+}
+
+template <typename T>
+void for_each_in_list(const goos::Object& list, T f) {
```

Comment:
I think this exists in `common/goos/ParseHelpers` already

Reply:
I can only find it in `goalc/compiler/compilation/Static.cpp`